content/courses/advanced-ada/parts/data_types/types_representation.rst

```diff
@@ -2775,6 +2775,238 @@ these array declarations are equivalent:
     end Shared_Var_Types;
 
 
+:ada:`Full_Access_Only`
+~~~~~~~~~~~~~~~~~~~~~~~
+
+A full access object is an object that requires that read or write operations
+on this object be performed by reading or writing all bits of the object (i.e.
+the *full object*) at once. Accordingly, a full access type is a type whose
```

Comment:
"at once."
I think this statement doesn't go far enough to distinguish such objects (and types) from atomic objects. The resulting dynamics semantics of these objects is that, even when the source code only reads or updates a part of the object, the entire object is read or updated (and the unchanged bits are written back). 
I believe the examples are also missing the above point.
See AARM22 C.6 {19.1/5 and 19.C/5}.
For examples, I can point you to some of the STM32 Ada Driver Library drivers, which lead to the request for this aspect. In these cases the STM device hardware reference manual requires volatile full access even when reading/updating subcomponents of the memory-mapped device.
The effect for an update is the same as if the source code does a read of the entire object into a temporary object, makes local subcomponent changes to that temporary, and then writes the temporary value back to the original object. For a read, the entire object is read into the temporary object, and then parts of the temporary are accessed individually. That's what the actual C version of the drivers actually does.  We didn't want to do that in Ada at the source code level so we have the compiler do it.

Reply:
Thanks for the clarification! I'll look into the Ada Driver Library for inspiration and rework the source-code examples later on.